src/csharp/Grpc.Core/RpcException.cs

```diff
@@ -44,6 +47,18 @@ public RpcException(Status status) : base(status.ToString())
         public RpcException(Status status, string message) : base(message)
         {
             this.status = status;
+            this.trailers = Metadata.Empty;
+        }
+
+        /// <summary>
+        /// Creates a new <c>RpcException</c> associated with given status and trailing response metadata.
+        /// </summary>
+        /// <param name="status">Resulting status of a call.</param>
+        /// <param name="trailers">Response trailing metadata.</param> 
+        public RpcException(Status status, Metadata trailers) : base(status.ToString())
```

Comment:
One thing that might keep the server-side simpler:

What if the `public RpcException(Status status, Metadata trailers)` constructor for `RpcException` was kept private, and only used within grpc library on client-side calls?
Then for server-side calls, the existing way of setting trailers could remain the only way.

- might lose consistency, but I think server-side changes here might not be needed